### PR TITLE
fix: missing update of upstream fragment and actor ids for table replacement

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As title.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by BugenZhao — COMMENTED** [review]
Does it mean that we'll get metadata corrupted when `CREATE SINK INTO TABLE` before this PR? Is there any tests that can cover this?

**Comment by yezizp2012** [comment]
> Does it mean that we'll get metadata corrupted when `CREATE SINK INTO TABLE` before this PR? 

I'm afraid yes, and also when `ALTER TABLE ADD/DROP COLUMN`. But these two fields are only used for scaling IIUC.

> Is there any tests that can cover this?

I'm afraid not, I will add some in scaling integration tests.

File: src/tests/simulation/tests/integration_tests/scale/table.rs [hunk 1]
```diff
@@ -85,3 +85,39 @@ async fn test_mv_on_scaled_table() -> Result<()> {
...
+    let fragment = cluster
+        .locate_one_fragment([identity_contains("materialize"), identity_contains("StreamTableScan")])
+        .await?;
+
+    cluster.reschedule_resolve_no_shuffle(fragment.reschedule([1], [0, 4])).await?;
+
+    let fragment = cluster
+        .locate_one_fragment([identity_contains("materialize"), identity_contains("union")])
+        .await?;
+    let (_, used) = fragment.parallel_unit_usage();
```
Comment (yezizp2012) [inline src/tests/simulation/tests/integration_tests/scale/table.rs:117]: Integration test added, previously the reschedule will take no effects and the information in system table is in-correct, which will fail the assertion.

**Review by BugenZhao — APPROVED** [review]
LGTM. THX!

**Pull request merged**